### PR TITLE
enable spreadtickets by default

### PR DESCRIPTION
**Pull request by raedah**

### CODE REVIEW

**Comment by alexlyp** [comment]
After testing @jrick pointed out that with the use of ticketbuyer.spreadticketpurchases as a bool flag, it is impossible to now set it "back" to off as an option:
bool flag '--ticketbuyer.spreadticketpurchases' cannot have an argument

Open to hear options on how to fix.

**Comment by alexlyp** [comment]
Ok after some discussion, this is what we came up with:

Add a new option to config.go for --nospreadticketpurchases and deprecate the existing --spreadticketpurchases

The default for --nospreadticketpurchases would be false (which is the equivalent of --spreadticketpurchases=true).  Then make sure the bools are correct for passing into ticketbuyer.

**Comment by alexlyp** [comment]
@raedah did you our comments about this?

**Comment by raedah** [comment]
I am back, ok, will do.

**Comment by raedah** [comment]
This tests well for me. Changes touch walletrpc api, so may require updates to decrediton.

**Comment by jrick** [comment]
This is failing travis:

```
# github.com/decred/dcrwallet/rpc/rpcserver
rpc/rpcserver/server.go:1586: unknown walletrpc.TicketBuyerConfigResponse field 'SpreadTicketPurchases' in struct literal
```

File: rpc/walletrpc/api.pb.go [hunk 1]
```diff
@@ -2411,26 +2411,26 @@ func (*TicketBuyerConfigRequest) ProtoMessage()               {}
...
+	FeeSource               string  `protobuf:"bytes,8,opt,name=fee_source,json=feeSource" json:"fee_source,omitempty"`
+	FeeTargetScaling        float64 `protobuf:"fixed64,9,opt,name=fee_target_scaling,json=feeTargetScaling" json:"fee_target_scaling,omitempty"`
+	MinFee                  int64   `protobuf:"varint,10,opt,name=min_fee,json=minFee" json:"min_fee,omitempty"`
+	MaxFee                  int64   `protobuf:"varint,12,opt,name=max_fee,json=maxFee" json:"max_fee,omitempty"`
+	MaxPerBlock             int64   `protobuf:"varint,13,opt,name=max_per_block,json=maxPerBlock" json:"max_per_block,omitempty"`
+	MaxPriceAbsolute        int64   `protobuf:"varint,14,opt,name=max_price_absolute,json=maxPriceAbsolute" json:"max_price_absolute,omitempty"`
+	MaxPriceRelative        float64 `protobuf:"fixed64,15,opt,name=max_price_relative,json=maxPriceRelative" json:"max_price_relative,omitempty"`
+	MaxInMempool            int64   `protobuf:"varint,17,opt,name=max_in_mempool,json=maxInMempool" json:"max_in_mempool,omitempty"`
+	NoSpreadTicketPurchases bool    `protobuf:"varint,20,opt,name=spread_ticket_purchases,json=noSpreadTicketPurchases" json:"no_spread_ticket_purchases,omitempty"`
```
Comment (alexlyp) [inline rpc/walletrpc/api.pb.go]: the generated name on this option stayed as spread_ticket_purchases.

@jrick any idea best way to fix that?

Comment (jrick) [inline rpc/walletrpc/api.pb.go]: it looks like this file wasn't autogenerated as it should have been and was manually edited instead.

The correct way of making changes to the grpc server is described at https://github.com/decred/dcrwallet/blob/master/rpc/documentation/serverchanges.md

**Comment by raedah** [comment]
resolved

**Comment by raedah** [comment]
@jrick please do take it over. All my focus is going into the new sdiff algo currently.

**Comment by raedah** [comment]
This has been fixed per the comments and passed manual testing.
api.pb is generated now, though I dont know why it still shows the old option
https://github.com/raedah/dcrwallet/blob/615f588a5fd1cd20befbcd798d7653f197d10ff0/rpc/walletrpc/api.pb.go#L2468

**Comment by alexlyp** [comment]
@raedah you need to update api.proto with the changes, then regenerate api.pb.go

With your current PR there would be no need to regenerate.

If you do want to update the grpc fo

**Comment by alexlyp** [comment]
Opened a new PR with these changes.  Closing this one.

**Pull request closed without merging**